pkg/secrethub/credentials.go

```diff
@@ -71,7 +73,31 @@ func (s credentialService) Create(creator credentials.Creator) error {
 	return nil
 }
 
-// ListMine lists all credentials of the currently authenticated account.
-func (s credentialService) ListMine() ([]*api.Credential, error) {
-	return s.client.httpClient.ListMyCredentials()
+type CredentialIterator interface {
+	Next() (api.Credential, error)
+}
+
+type credentialIterator struct {
+	credentials  []*api.Credential
+	currentIndex int
+}
+
+func (c *credentialIterator) Next() (api.Credential, error) {
+	currentIndex := c.currentIndex
+	if currentIndex >= len(c.credentials) {
+		return api.Credential{}, iterator.Done
+	}
+	c.currentIndex++
+	return *c.credentials[currentIndex], nil
+}
+
+// List lists all credentials of the currently authenticated account.
+func (s credentialService) List() (CredentialIterator, error) {
```

Comment:
To be consistent with audit events and for simplicity for the user, creating the iterator shouldn't return an error.
We can fetch the list of credentials on the first call to `Next`.